test/unit/src/types/SendCachedAppDataReq.test.ts

```diff
@@ -0,0 +1,168 @@
+import { VectorBufferStream } from '../../../../src'
+import { TypeIdentifierEnum } from '../../../../src/types/enum/TypeIdentifierEnum'
+import { initAjvSchemas } from '../../../../src/types/ajv/Helpers'
+import { stateManager } from '../../../../src/p2p/Context'
+import { AppObjEnum } from '../../../../src/types/enum/AppObjEnum'
+import { Utils } from '@shardus/types'
+
+import {
+  SendCachedAppDataReq,
+  serializeSendCachedAppDataReq,
+  deserializeSendCachedAppDataReq,
+} from '../../../../src/types/SendCachedAppDataReq'
+import { deserialize } from 'v8'
+
+// Mock the Context module and its nested structure
+jest.mock('../../../../src/p2p/Context', () => ({
+    setDefaultConfigs: jest.fn(),
+    stateManager: {
+      app: {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        binarySerializeObject: jest.fn((enumType: AppObjEnum, data: any) => Buffer.from(Utils.safeStringify(data), 'utf8')),
+        binaryDeserializeObject: jest.fn((enumType: AppObjEnum, buffer: Buffer) => Utils.safeJsonParse(buffer.toString('utf8'))),
+      },
+    },
+  }))
+
+const cSendCachedAppDataReqVersion = 1
+
+describe('SendCachedAppDataReq Serialization and Deserialization', () => {
+    beforeAll(() => {
+        initAjvSchemas()
+    })
+
+    describe('serializeSendCachedAppDataReq Serialization', () => {
+        test('should serialize with root true', () => {
+            const obj: SendCachedAppDataReq = {
```

Comment:
note, all unit tests pass except for this one. starts with a compile error on this line.